src/config/finalizeConfiguration.jsx

```diff
@@ -0,0 +1,24 @@
+import { gridTeaserDisableStylingSchema } from '@plone/volto/components/manage/Blocks/Teaser/schema';
+
+/**
+ * finalizeCoreConfiguration
+ *
+ * It allows the finalization of the configuration object. In some edge cases, the core need to
+ * modify the configuration **after** the add-ons apply theirs.
+ *
+ * @export
+ * @param {*} config
+ */
+export function finalizeCoreConfiguration(config) {
+  // This is required in order to initialize the inner blocksConfig
+  // for the grid block, since we need to modify how the inner teaser
+  // block behave in it (= no schemaEnhancer fields for teasers inside a grid)
+  // Afterwards, it can be further customized in add-ons using the same technique.
+  config.blocks.blocksConfig.gridBlock.blocksConfig = {
+    ...config.blocks.blocksConfig,
+  };
+  config.blocks.blocksConfig.gridBlock.blocksConfig.teaser = {
+    ...config.blocks.blocksConfig.teaser,
+    schemaEnhancer: gridTeaserDisableStylingSchema,
```

Comment:
@sneridagh Doesn't this make it impossible for an addon to use custom schema enhancers for the teaser block inside a grid?

Reply:
Yes, you are correct... I've been struggling with this one... because slate is an add-on, we need their config first, if not, then the slate block is not in the `blockConfig` when you override it in the Grid, then it's not available. A chicken-egg situation :( not sure how to deal with it. 

Maybe initializing it from the core? Any ideas? /cc @tiberiuichim

Reply:
Ok, I think I got it, incoming commit...